### PR TITLE
Enhancement to consider subtask bug which are not tagged in release.

### PR DESCRIPTION
**Pull request by aksshriv1**

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java [hunk 1]
```diff
@@ -79,6 +79,7 @@
 @Slf4j
 public class JiraServiceR {
 
+	public static final String BUG = "Bug";
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java:82]: use variable from NormalizedJira

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java [hunk 2]
```diff
@@ -385,6 +396,18 @@ public void fetchJiraIssueReleaseForProject(String basicProjectConfigId, String
 		}
 	}
 
+	private Set<JiraIssue> fetchSubTaskDefectsRelease(String projectConfigId, Set<String> storyIDs) {
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java:399]: add documentation and comments in all methods

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java [hunk 3]
```diff
@@ -385,6 +396,18 @@ public void fetchJiraIssueReleaseForProject(String basicProjectConfigId, String
...
+		ObjectId basicProjectConfigId = new ObjectId(projectConfigId);
+		FieldMapping fieldMapping = configHelperService.getFieldMappingMap().get(basicProjectConfigId);
+		if (CollectionUtils.isNotEmpty(storyIDs) && fieldMapping != null
+				&& CollectionUtils.isNotEmpty(fieldMapping.getJiraSubTaskDefectType())) {
+			return jiraIssueRepository
+					.findByBasicProjectConfigIdAndDefectStoryIDInAndOriginalTypeIn(projectConfigId, storyIDs,
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java:405]: have you check OriginalType is available in old existing data also?

**Review by eswbogol — DISMISSED** [review]

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java (hunk 3 above)
Comment (aksshriv1) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java:405]: Yes

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java (hunk 1 above)
Comment (aksshriv1) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java:82]: Done

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java (hunk 2 above)
Comment (aksshriv1) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/service/JiraServiceR.java:399]: done

**Review by Chittauri — APPROVED** [review]

**Review by eswbogol — APPROVED** [review]

**Pull request merged**